core/src/stages/generate_pose.cpp

```diff
@@ -40,13 +40,47 @@
 #include <moveit/planning_scene/planning_scene.h>
 #include <rviz_marker_tools/marker_creation.h>
 
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#include <tf2/convert.h>
+
+#include <chrono>
+
+namespace {
+// TODO(henningkayser): support user-defined random number engines
+std::random_device rd_;
+std::mt19937 gen_(rd_());
+}
+
 namespace moveit {
 namespace task_constructor {
 namespace stages {
 
 GeneratePose::GeneratePose(const std::string& name) : MonitoringGenerator(name) {
 	auto& p = properties();
 	p.declare<geometry_msgs::PoseStamped>("pose", "target pose to pass on in spawned states");
+	p.declare<size_t>("max_solutions", 20,
+	                  "limit of the number of spawned solution in case randomized sampling is enabled");
+	p.property("timeout").setDefaultValue(10.0 /* seconds */);
+}
```

Comment:
I guess it would make sense to set the random distributions from parameters?
Did you attempt to do that?

Reply:
I somehow found it weird to store functions or distribution instances as parameters. Do we plan to share them? Otherwise, we could think about just storing meta-information (like instance type + parameter value) and initialize the samplers in `compute()`

Reply:
Yes, I was thinking about distribution specifications, not about the C++ instances.

Reply:
I imagine it'd be difficult to add support for distribution functions if the number/type of params is unknown at compile time.

Reply:
But they *are* known, because `SampleRandomPose`, or whatever Henning will call the derived class :wink: , needs to have support for it compiled in.

Reply:
Ignore my comment. This is fine as-is.